src/Bicep.Core/Syntax/SyntaxFactory.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Bicep.Core.Extensions;
+using Bicep.Core.Parsing;
+
+namespace Bicep.Core.Syntax
+{
+    public static class SyntaxFactory

```

Comment:
We now have multiple classes serving the same purpose in different projects:
- Bicep.Core.Syntax.SyntaxFactory
- Bicep.Decompiler.BicepHelpers.SyntaxHelpers
- Bicep.Core.UnitTests.Utils.TestSyntaxFactory

To make this PR less overwhelming, I'll send a separate PR to reduce them to one class.